app/src/main/java/io/homeassistant/companion/android/notifications/MessagingManager.kt

```diff
@@ -1287,6 +1288,18 @@ class MessagingManager @Inject constructor(
             return@withContext FileProvider.getUriForFile(context, "${context.packageName}.provider", file)
         }
 
+    private suspend fun isGif(url: URL): Boolean {
+        try {
+            val connection = url.openConnection()
+            connection.setRequestProperty("User-Agent", HomeAssistantApis.USER_AGENT_STRING)
+            val contentType = connection.contentType
```

Comment:
It looks like this results in the entire image (or whatever is behind the URL) being downloaded. As the image is also fetched later, that is work that should be avoided as it duplicates data used. Try only getting the headers which should contain the content type. This also doesn't handle the authentication required images (hosted on the HA server).

Reply:
You are right. I commit a "connection.requestMethod = "HEAD" for that. 
But what du you mean with "authentication required images (hosted on the HA server)"? I only know direct image paths but these ones should be handled with the filename ending.
Do you maybe mean the use of Image entities? For that this really does not work. But i'm not shure how to resolve this problem. Maybe we could check the file header here?

Reply:
I now added a byte checkup for the Entity urls. Im not sure if this works good like this, but i will check it with the generated apk!

Reply:
Because I dont have a image entitiy and i dont now how to add one manually i can't test the byte GIF-check with the image entity. I tryed it with the camera entity, because i used a gif in my camera entity but that doesn't work. Do you now, if the camera entity converts the input somehow? If yes i will delete the camera gif-check.